python/cuspatial/cuspatial/core/_column/geocolumn.py

```diff
@@ -130,10 +131,11 @@ def __init__(
             self.polygons.name = "polygons"
         else:
             raise TypeError("All four Tuple arguments must be cudf.ListSeries")
-        base = cudf.core.column.column.arange(0, len(self), dtype="int32").data
-        super().__init__(base, size=len(self), dtype="int32")
+        super().__init__(None, size=len(self), dtype="geometry")
         if shuffle_order is not None:
-            self._data = shuffle_order
+            self._gather_map = shuffle_order
```

Comment:
Ok... I see where it's going. `GeoColumn` wants to maintain an intrinsic order that's different from the data's physical layout. I can think of both side of pros and cons. But perhaps it's the best to let you explain the reasons behind this design decision.

Reply:
And it's an important detail that we should mention in `libary_design`.

Reply:
Let me see how much of this I can recall. I needed to be taking notes during this process.

At first I wrote `GeoSeries.memory_usage`, but noticed that I needed to support `GeoDataFrame.memory_usage`. It is inherited from `DataFrame`, but was failing because `DataFrame.memory_usage` calls the `Series._from_data` function, which was failing because, I think, the `base` data as `int32` wasn't compatible. I'm going to see if I can reproduce the failure case that lead to this rabbit holing. It's possible it should be separated into another PR, but I don't think so.

Reply:
Ohhhh haha. What you're looking at here is part of #709, which I incorrectly said was #585 in the description. How about you review #709 first, then after merging it this should be simple again.

Reply:
In #585, using `super()._init_from_dict` like causes a host of internal `cudf.DataFrame` methods to be called.